File: conformance/tests/bounded-loop/bounded-loop.runar.zig
const runar = @import("runar");

pub const BoundedLoop = struct {
    pub const Contract = runar.SmartContract;

    expectedSum: i64,

    pub fn init(expectedSum: i64) BoundedLoop {
        return .{ .expectedSum = expectedSum };
    }

    pub fn verify(self: *const BoundedLoop, start: i64) void {
        var sum: i64 = 0;
        var i: i64 = 0;
        while (i < 5) : (i += 1) {
            sum = sum + start + i;
        }
        runar.assert(sum == self.expectedSum);
    }
};

File: conformance/tests/bounded-loop/bounded-loop.runar.go
//go:build ignore

package contract

import "runar"

type BoundedLoop struct {
	runar.SmartContract
	ExpectedSum runar.Int `runar:"readonly"`
}

func (c *BoundedLoop) Verify(start runar.Int) {
	sum := runar.Int(0)
	for i := runar.Int(0); i < 5; i++ {
		sum = sum + start + i
	}
	runar.Assert(sum == c.ExpectedSum)
}
